code/iaas/ha/src/main/java/io/cattle/platform/ha/monitor/dao/impl/PingInstancesMonitorDaoImpl.java

```diff
@@ -4,18 +4,21 @@
 import static io.cattle.platform.core.model.tables.HostTable.*;
 import static io.cattle.platform.core.model.tables.InstanceHostMapTable.*;
 import static io.cattle.platform.core.model.tables.InstanceTable.*;
-import io.cattle.platform.core.constants.InstanceConstants;
+import io.cattle.platform.core.constants.CommonStatesConstants;
+import io.cattle.platform.core.model.Instance;
 import io.cattle.platform.db.jooq.dao.impl.AbstractJooqDao;
 import io.cattle.platform.ha.monitor.dao.PingInstancesMonitorDao;
 
+import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 
 public class PingInstancesMonitorDaoImpl extends AbstractJooqDao implements PingInstancesMonitorDao {
 
     @Override
-    public Map<String, String> getInstances(long agentId) {
-        Map<String, String> result = create()
-                .select(INSTANCE.UUID, INSTANCE.EXTERNAL_ID)
+    public Map<String, Instance> getInstances(long agentId) {
+        List<Instance> instances = create()
```

Comment:
You are right. I was thinking about the network agent, dropping this question :)